app/models/payment_allocation_set.rb

```diff
@@ -1,5 +1,6 @@
 class PaymentAllocationSet < ApplicationRecord
   belongs_to :customer
+  before_validation :filter_zero_percent_payment_allocations
```

Comment:
Consider just looking at this scope in the validador. This will cause a subtle hard to debug bug in the future.